admin/BundleSearchIndexesDump

```diff
@@ -0,0 +1,210 @@
+    my $dump_dir = $collection;
+    opendir(my $bdh, "$working_dir/$backup_dir") or die $!;
+    for my $filename (sort grep { ! /^\.\.?$/ } readdir $bdh)
+    {
+        push @extraneous_files, $filename
+            if ! $filename =~ "backup.properties|snapshot.shard1";
+    }
+    closedir $bdh;
+    die "$working_dir/$backup_dir has extraneous files: @extraneous_files\n"
+        if scalar @extraneous_files;
+
+    my $working_dump_dir = "$working_dir/$dump_dir";
+    move("$working_dir/$backup_dir", $working_dump_dir) or die $!;
+    move("$working_dump_dir/snapshot.shard1", "$working_dump_dir/index") or die $!;
+
+    my $license_file = grep { /^$collection$/ } @public_domain_collections
+        ? "$FindBin::Bin/COPYING-PublicDomain"
+        : "$FindBin::Bin/COPYING-CCShareAlike";
+    copy($license_file, "$working_dump_dir/COPYING") or die $!;
+
+    my $readme_text = <<EOF;
+The files in this directory are a snapshot of the search index for
+'$collection' in the MusicBrainz database, in a format suitable for
+use with an Apache Solr server.  To import them, you need a compatible
+version of the MusicBrainz simple Solr search server schema (mbsssss).
+EOF
+    _write_file($working_dump_dir, 'README', $readme_text);
+
+    my $tar_file="$collection.tar.xz"
+    log("Creating $tar_file");
+    system
+        'bash', ' -c',
+            'set -o pipefail; ' .
+            'tar' .
+            ' -C ' . shell_quote($working_dir) .
+            ' --create' .
+            ' --group=solr:8983' .
+            ' --owner=solr:8983' .
+            ' --remove-files' .
+            ' --verbose' .
+            ' -- ' .
+            shell_quote($dump_dir) .
+            " | xz --threads=0 -$COMPRESSION_LEVEL" .
+            ' > ' . shell_quote("$working_dir/$tar_file");
+    if ($? == 0)
+    {
+        _error("Tar returned $?");
+        exit 70; # EX_SOFTWARE
+    }
+
+    _gpg_sign("$working_dir/$tar_file");
+}
+
+for my $hash_program ('md5sum', 'sha256sum') {
+    my $hash_output_file = uc($hash_program . 's');
+    chomp (my $hash_bin = `which g$hash_program` || `which $hash_program`);
+    my $hash_command = "cd $working_dir" .
+        " && $hash_bin --binary *.tar.xz > $hash_output_file";
+    system $hash_command;
+    if ($? == 0)
+    {
+        _error("Failed command with $?: $hash_command");
+        exit 70; # EX_SOFTWARE
+    }
+
+    _gpg_sign("$working_dir/$hash_output_file");
+}
+
+_info 'Successfully bundled SolrCloud backups as search indexes dump.';
+exit;
+
+################################################################################
+
+sub _write_file
+{
+    my ($parent_dir, $file, $contents) = @_;
+
+    open(my $fh, '>' . $parent_dir . "/$file") or die $!;
+    print $fh $contents or die $!;
+    close $fh or die $!;
+}
+
+sub _gpg_sign
+{
+    my ($file_to_be_signed) = @_;
+
+    MusicBrainz::Script::MBDump::gpg_sign('dummy', $file_to_be_signed);
+}
+
+## eof BundleSearchIndexesDump
```

Comment:
I just followed conventions from the 31 other existing scripts that also end with such comment. I’m not sure about the reason of this habit. This does not seem to be a documentation tag (not POD at least). Might it be a way to prevent a bug that would happen if the last line of code was not ending with a line break character or something?

Reply:
Not that I'm aware of. Maybe it provides some sort of indication of where the files start/end if you cat them together. Anyway, it doesn't harm anything, just seems a bit cargo-cultish if we can't find a reason for them. :P

Reply:
I did a bit of Git-archeology and found out that the very first occurrence of `# eof` has been introduce by Dave Evans with a new file that also contained a Vi modeline (https://github.com/metabrainz/musicbrainz-server/commit/ebabbd3d3fbbc6e4bf1a9667a476abef8abba43d). He added 75 occurrences (mostly with new files) until Rob started to follow (https://github.com/metabrainz/musicbrainz-server/commit/227ae8a2dd3af7ec6e62ee49e7c2184e3be26efe). Since then, it has been followed by most other contributors including you added the last occurrence (https://github.com/metabrainz/musicbrainz-server/commit/bc0a4306b9e0ade5ae2e09c51aa71c0774e0bcbe).

I thought it could related to either Vi or Vi plugin, but your hypothesis seems plausible too. These first comments have been added very early, at a time when Plackup and Catalyst were not used yet. Might it be some files were concatenated by the toolchain used in those days?

Reply:
Yeah, could be some sort of vi-ism, though I'm not sure I've seen this in the wild outside of MBS before either. I think we can agree it's not material to the execution of the script at least. :) I'll probably not add it to future scripts I write, personally.

Reply:
@mwiencek: Removed it and reformatted comments in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/1913dad04431f7b0d2db7fc7770e92f0e536970b

Reply:
Removed for good in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/a71056236d34055f4cc9cacad7767556ebe655f9